airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -244,6 +244,10 @@ def get_report_date_ranges(start_report_date: Optional[datetime]) -> Iterable[st
         now = datetime.utcnow()
         if not start_report_date:
             start_report_date = now
+        
+        # You cannot pull data for amazon ads more than 60 days
+        if (now - start_report_date).days > 59:
```

Comment:
Since I am looking back for 3 days in the `stream_slices` function if the `start_date` here in this function comes to be older than 60 days then it's gonna adjust it in `get_report_date_ranges`

Reply:
Yeah but for example if you have start_report_date that is 59 days before today, you will substract 3 days and it would be 62 days which is out of timerange window. So it should be 59 - LOOK_BACK_WINDOW

Reply:
Made the change @avida